src/WebUI/Workloads/WorkloadsStore.ts

```diff
@@ -80,24 +84,40 @@ export class WorkloadsStore extends StoreBase {
     }
 
     private _setReplicaSetsList = (replicaSetList: V1ReplicaSetList): void => {
-        this._state.replicaSetList = replicaSetList;
-        this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        const _stateItems = this._state.replicaSetList ? this._state.replicaSetList.items : [];
+        const _replicaItems = replicaSetList ? replicaSetList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _replicaItems);
+        if (!isEquals || !this._state.replicaSetList) {
+            this._state.replicaSetList = replicaSetList;
+            this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        }
         if (this._state.replicaSetList && this._state.replicaSetList.items && this._state.replicaSetList.items.length > 0) {
             this.emit(WorkloadsEvents.WorkloadsFoundEvent, this);
         }
     }
 
     private _setDaemonSetsList = (daemonSetList: V1DaemonSetList): void => {
-        this._state.daemonSetList = daemonSetList;
-        this.emit(WorkloadsEvents.DaemonSetsFetchedEvent, this);
+        const _stateItems = this._state.daemonSetList ? this._state.daemonSetList.items : [];
+        const _daemonSetItems = daemonSetList ? daemonSetList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _daemonSetItems);
+        if (!isEquals || !this._state.daemonSetList) {
+            this._state.daemonSetList = daemonSetList;
+            this.emit(WorkloadsEvents.DaemonSetsFetchedEvent, this);
+        }
         if (this._state.daemonSetList && this._state.daemonSetList.items && this._state.daemonSetList.items.length > 0) {
             this.emit(WorkloadsEvents.WorkloadsFoundEvent, this);
         }
     }
 
     private _setStatefulsetsList = (statefulSetList: V1StatefulSetList): void => {
-        this._state.statefulSetList = statefulSetList;
-        this.emit(WorkloadsEvents.StatefulSetsFetchedEvent, this);
+        const _stateItems = this._state.statefulSetList ? this._state.statefulSetList.items : [];
+        const _stateFulSetItems = statefulSetList ? statefulSetList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _stateFulSetItems);
+        //if this is the first call or empty data
+        if (!isEquals || !this._state.statefulSetList) {
+            this._state.statefulSetList = statefulSetList;
+            this.emit(WorkloadsEvents.StatefulSetsFetchedEvent, this);
+        }
         if (this._state.statefulSetList && this._state.statefulSetList.items && this._state.statefulSetList.items.length > 0) {
```

Comment:
should not we keep this inside the if condition of "if (!isEquals || !this._state.statefulSetList) {"?